tensornetwork/backends/numpy/numpy_backend_test.py

```diff
@@ -419,6 +419,45 @@ def test_eigsh_lanczos_raises():
     backend.eigsh_lanczos(lambda x: x, initial_state=[1, 2, 3])
 
 
+def test_gmres_raises():
+  backend = numpy_backend.NumPyBackend()
+  dummy_mv = lambda x: x
+  N = 10
+  with pytest.raises(ValueError): # x0, b have different sizes
```

Comment:
add pytest.raise(ValueError, match="the error string") to all checks so we know the right error is raised